lib/internal/process/promises.js

```diff
@@ -26,6 +33,7 @@ function setupPromises(scheduleMicrotasks) {
     if (hasBeenNotified !== undefined) {
       hasBeenNotifiedProperty.delete(promise);
       if (hasBeenNotified === true) {
+        promiseInternals.untrackPromise(promise);
```

Comment:
We should only be creating a `Persistent` for rejected promises that aren't handled within a tick. If it's happening for all then the PR needs to be updated.